### ISSUE TITLE
certificate_p12 Output (pfx) not usable on Windows

### ISSUE THREAD

**Issue by e-bits** [vancluever/terraform-provider-acme#381]
Im running acme provider version 2.20.0 and cannot get an output of the p12 cert working on my windows machine. 

I tried the same with lego cli directly and it is working as expected.

Here my lego cli command:
`lego --accept-tos --server=https://acme-staging-v02.api.letsencrypt.org/directory --pfx --pfx.pass "myVerySecert1234" --domains "sub.example.com" --domains "*.sub.example.com" --email "sub-example@mailinator.com" --dns "azuredns" --dns.disable-cp run`

Here is my terraform:
```
provider "acme" {
  server_url = "https://acme-staging-v02.api.letsencrypt.org/directory"
}

resource "tls_private_key" "private_key" {
  algorithm   = "ECDSA"
  ecdsa_curve = "P384"
}

resource "acme_registration" "reg" {
  account_key_pem = tls_private_key.private_key.private_key_pem
  email_address   = "sub-example@mailinator.com"
}

resource "acme_certificate" "certificate" {
  account_key_pem              = acme_registration.reg.account_key_pem
  common_name                  = "sub.example.com"
  subject_alternative_names    = ["*.sub.example.com"]
  key_type                     = "P384"
  disable_complete_propagation = true
  certificate_p12_password     = "myVerySecert1234"

  dns_challenge {
    provider = "azuredns"
  }
}

resource "local_file" "result" {
  filename       = "${path.module}/sub.example.com.pfx"
  content_base64 = acme_certificate.certificate.certificate_p12
}
```

If I try to load the generated `sub.example.com.pfx` in Windows I get a Failure. Also in [XCA ](https://hohnstaedt.de/xca/) the pfx coming from Terraform can't be loaded. I can't find any detailed Error Messages to provide here.

Can anyone help here?

**Comment by vancluever** [comment]
@e-bits do you have any other detail here, error messages, etc?

**Comment by e-bits** [comment]
On Windows I try to open the generated pfx with the following default behaviour of Windows with:
`%SystemRoot%\system32\rundll32.exe cryptext.dll`

Therefore I receive the following error:
![2024-02-05 22_28_46-Invalid Public Key Security Object File](https://github.com/vancluever/terraform-provider-acme/assets/5858471/d69a6c88-2c9c-4aa8-80de-6aaf0b624b14)

If I try to analyze the generated PFX with the following command: `openssl pkcs12 -info -in sub.example.com.pfx`

I get asked for the password twice, once with a prompt `Enter Import Password` and afterwards `Enter PEM pass phrase`. I compared the Output from lego generated PFX to terraform generated PFX and everything looks good.

It feels like a Linux Windows Issue with LF or CRLF. Because if I manually upload the terraform generated PFX to Azure Key Vault it get's accepted there and loads properly.

I wish I could provide an EventLog or similar but I can't find anything on my Windows machine. 😞

**Comment by vancluever** [comment]
@e-bits that's interesting. Can you decode the file yourself from base64 to binary without Terraform to see what happens?

**Comment by e-bits** [comment]
Hi @vancluever, sorry for letting you wait. I was on vacation and not able to decode.

To manually decode I did the following. I took the attribute `certificate_p12` from the Terraform statefile and decoded it with this command on Windows:
`certutil -decode encodedInputFileName decodedOutputFileName`

If I compare the 2 pfx, one generate by acme-provider and the other one manually decoded, they are binary same.

I think the failure is related to my machine. I will do some additional tests and will keep you posted.
